src/game/Game.ts

```diff
@@ -0,0 +1,128 @@
+import KeyListener from "./KeyListener";
+
+enum Tile {
+    FLOOR,
+    WALL,
+    UNDEFINED,
+}
+
+interface PlayerObject {
+    r: number;
+    c: number;
+}
+
+export default class Game {
+    map: Tile[][];
+
+    player: PlayerObject;
+
+    private keyListener = new KeyListener();
+
+    constructor(map: string) {
+        this.loadMapFromText(map);
+        console.log(this.textRender(this.map, this.player));
+    }
+
+    step(time: DOMHighResTimeStamp) {
+        if (this.keyListener.anyPressed()) {
+            const inputs = this.keyListener.inputs;
+
+            // Attempt to move character
+            const previousR = this.player.r;
+            const previousC = this.player.c;
+
+            if (inputs.moveUp) this.player.r--;
+            else if (inputs.moveDown) this.player.r++;
+            else if (inputs.moveLeft) this.player.c--;
+            else if (inputs.moveRight) this.player.c++;
+
+            if (this.collision(this.player.r, this.player.c)) {
+                this.player.r = previousR;
+                this.player.c = previousC;
+            }
+
+            // Print map
+            console.log(this.textRender(this.map, this.player));
+
+            this.keyListener.clear();
+        }
```

Comment:
Refactor to use a guard clause here to prevent unnecessary indentation:
```js
step(time: DOMHighResTimeStamp) {
  if (!this.keyListener.anyPressed()) {
    return;
  }
  const inputs = this.keyListener.inputs;

  // Attempt to move character
  const previousR = this.player.r;
  const previousC = this.player.c;
  ...
}
```